### PR TITLE
#716 - new example in Enum.4

### PR DESCRIPTION
**Pull request by robert-andrzejuk**
Old example causes stack overflow (inifinite recursion call to operator++)

### CODE REVIEW

**Comment by kurman-vladimir** [comment]
Shouldn't we return here a reference?

**Comment by robert-andrzejuk** [comment]
All checks have failed
... Hmm I wonder what is wrong?

    make[1]: *** [cpplint-all] Error 1
    
    make[1]: Leaving directory `/home/travis/build/isocpp/CppCoreGuidelines/scripts/build/codeblocks'
    
    make: *** [cpplint-all] Error 2
    
    ##################### Bad chars check ##################
    
    make: Target `default' not remade because of errors.
    
    The command "cd scripts; make -k" exited with 2.

**Comment by cubbimew** [comment]
@robert-andrzejuk  scroll up to where it says

```
Renum-oper0.cpp:27:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
Done processing Renum-oper0.cpp
Total errors found: 1
 14  // ../CppCoreGuidelines.md : 8389
```

**Comment by robert-andrzejuk** [comment]
:-D Thanks! that did it :-)

**Comment by gdr-at-ms** [comment]
Squash?

**Comment by robert-andrzejuk** [comment]
Squashed.

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -8388,11 +8388,21 @@ Convenience of use and avoidance of errors.
...
+        case Day::wed: return (d = Day::thu);
+        case Day::thu: return (d = Day::fri);
+        case Day::fri: return (d = Day::sat);
+        case Day::sat: return (d = Day::sun);
+        default: // case Day::sun:
+            return (d = Day::mon);
+        }
     }
-
+
```
Comment (blakehawkins) [inline CppCoreGuidelines.md]: you have EOL whitespace

Comment (robert-andrzejuk) [inline CppCoreGuidelines.md]: It's in a code block. Remove it?

**Comment by robert-andrzejuk** [comment]
Removed EOL whitespace

**Comment by d25fe0be** [comment]
`Day tomorrow = ++today;` in the original example effectively makes `today` equal to `tomorrow`, could / should we make a change to this as well?

**Comment by robert-andrzejuk** [comment]
Extended the example:
- added a while loop which demonstrates the use of the increment operator, 
- added a function "after" to remove the issue from above comment.

**Comment by d25fe0be** [comment]
Now that we illustrated the usage of `operator ++` by:
```
Day some_day = Day::wed;

while (some_day != Day::sat)
    ++some_day;
```
Maybe we could remove `today` and `tomorrow` entirely? (As they're no longer related to defining operators on enumerations.)

**Comment by robert-andrzejuk** [comment]
The  title is **Enum.4: Define operations on enumerations for safe and simple use**
So not only operators.

I think the function _after_ is also a good example.

I'll leave it to the scrutiny of the editors.

**Comment by d25fe0be** [comment]
My bad. I misread the title. :)

**Comment by AndrewPardoe** [comment]
Thank you for the suggested fix on this. We discussed it, and Bjarne had a different change he wanted to make to the sample. We're tracking this in your issue #716.

**Pull request closed without merging**